content/docs/rfcs/62/README.md

```diff
@@ -0,0 +1,1077 @@
+    ROLE_NONE = 0;
+    ROLE_OWNER = 1;
+    ROLE_ADMIN = 4;
+    ROLE_TOKEN_MASTER = 5;
+  }
+  repeated Roles roles = 1;
+  repeated RevealedAccount revealed_accounts = 2 [deprecated = true];
+  uint64 last_update_clock = 3;
+}
+
+message CommunityPermissions {
+  enum Access {
+    UNKNOWN_ACCESS = 0;
+    NO_MEMBERSHIP = 1;
+    INVITATION_ONLY = 2;
+    ON_REQUEST = 3;
+  }
+
+  bool ens_only = 1;
+  // https://gitlab.matrix.org/matrix-org/olm/blob/master/docs/megolm.md is a candidate for the algorithm to be used in case we want to have private communityal chats, lighter than pairwise encryption using the DR, less secure, but more efficient for large number of participants
+  bool private = 2;
+  Access access = 3;
+}
+```
+
+#### Payload
+
+| Field | Name | Type | Description |
+| ----- | ---- | ---- | ---- |
+| 1 | clock | `uint64` | Clock value of the message | 
+| 2| members | `map<string, CommunityMember>` | The members of the community |
+| 3 | permissions | `CommunityPermissions` | Image data associated with the chat key |
+| 4 | display_name | `string` | The self-assigned display_name of the chat key |
+| 5 | description | `string` | The description of the chat |
+| 6 | color | `string` | The color of the chat |
+| 7 | emoji | `string` | The emoji of the chat |
+| 8 | social_links | `array<SocialLink>` | A list of links to social platforms |
+| 9 | first_message_timestamp | `uint32` | First known message timestamp in seconds |
+
+### CommunityRequestToJoin
+
+A `CommunityRequestToJoin` represents a request to join a community, sent by a user that is not yet a member of that community. 
+A request to join a community includes a list of `RevealedAccount`.
+These are wallet addresses that users are willing to reveal with the community's control node and admins.
+
+```protobuf
+message CommunityRequestToJoin {
+  uint64 clock = 1;
+  string ens_name = 2;
+  string chat_id = 3;
+  bytes community_id = 4;
+  string display_name = 5;
+  repeated RevealedAccount revealed_accounts = 6;
+}
+
+message RevealedAccount {
+  string address = 1;
+  bytes signature = 2;
+  repeated uint64 chain_ids = 3;
+  bool isAirdropAddress = 4;
+}
+```
+
+#### Payload
+
+| Field | Name | Type | Description |
+| ----- | ---- | ---- | ---- |
+| 1 | clock | `uint64` | Clock value of the message | 
+| 2| ens_name | `string` | The ENS of the user sending the request |
+| 3 | chat_id | `string` | The id of the chat to request access to |
+| 4 | community_id | `bytes` | The public key of the community |
+| 5 | display_name | `string` | The display name of the usre sending the request |
+| 6 | revealed_accounts | `array<RevealedAccount>` | A list of accounts to reveal to the control node |
+
+### PinMessage
+
+A `PinMessage` is a signal that tells a client that a specific message has to be marked as pinned.
+
+```protobuf
+message PinMessage {
+  uint64 clock = 1;
+  string message_id = 2;
+  string chat_id = 3;
+  bool pinned = 4;
+  MessageType message_type = 5;
+}
```

Comment:
are there permissions to pin a message, and if so, please describe behaviour of clients to reject a pin from someone who cannot?